### ISSUE TITLE
Request: Support for debugging multiple Lua-states in the same program at once

### ISSUE THREAD

**Issue by Silverlan** [pkulchenko/ZeroBraneStudio#769]
My multiplayer game engine has two Lua-states, one handles serverside logic, the other everything clientside.
If a listen server or single-player game is started, both states are running in parallel. It would help immensely if it was possible to debug both states at the same time, without having to close and re-establish the connection every time to switch between them.
Right now ZeroBrane will just refuse any additional session requests: "_Refused a request to start a new debugging session as there is one in progress already._"

**Comment by pkulchenko** [comment]
@Silverlan, that's correct, only one debugging session is supported, but I do have a commercial extension that supports multiple debugging sessions and provides a way to switch between them, shows their current status (suspended/running), allows to set their names shown in the IDE and so on. I'll add a documentation page with details. You can contact me by email for pricing and licensing information.

**Comment by madmaxoft** [comment]
It might also be possible, although not super-comfortable, to run two instances of ZBS, each listening on a different debugging port, and each of the Lua states connecting to the specific port.

**Comment by pkulchenko** [comment]
@madmaxoft, this is a good point. @Silverlan, here is the FAQ item on how it can be done: https://studio.zerobrane.com/doc-faq#how-do-i-start-two-zerobrane-studio-instances-to-debug-two-applications-at-the-same-time, but you will need to configure different states to use different port numbers.

**Comment by Silverlan** [comment]
Sadly the commercial extension isn't an option for me, since I don't actually need this for myself, I need it for other people working on the engine.
I'll give the multi-instance option a try, thanks for the tip!

**Comment by myonlylonely** [comment]
@pkulchenko  Hi, does your commercial extension support states which are dynamically created by `luaL_newstate`? If this is the case, I would be very interested. I have just sent an email to you.

**Comment by madmaxoft** [comment]
@myonlylonely I believe you wanted to tag @pkulchenko instead :)

**Comment by pkulchenko** [comment]
Thanks, @madmaxoft! @myonlylonely, yes it does allow debugging states that are created with luaL_newstate. I'll respond to your email.

**Comment by myonlylonely** [comment]
@madmaxoft Thanks for the correction :), I have corrected it.

**Comment by myonlylonely** [comment]
@pkulchenko Hi, I still haven't got your response, is the email address on your GitHub page I should send to?

**Comment by pkulchenko** [comment]
I have the email and will respond shortly.
